pkg/sdk/metrics.go

```diff
@@ -21,23 +21,25 @@ import (
 	k8sutil "github.com/operator-framework/operator-sdk/pkg/util/k8sutil"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 )
 
 // ExposeMetricsPort generate a Kubernetes Service to expose metrics port
-func ExposeMetricsPort() {
+func ExposeMetricsPort() *v1.Service {
```

Comment:
If we are changing the signature, do we maybe want to add the ability to report an error?

Reply:
I am fine with that, but up to others.

Reply:
We can return an error here. 

But if we return an error on failing to initialize the service object when we're running locally then `operator-sdk up local` would always fail since `OPERATOR_NAME` is not set:
```Go
func ExposeMetricsPort() (*v1.Service, error) {
...
	service, err := k8sutil.InitOperatorService()
	if err != nil {
		logrus.Errorf("failed to initialize service object for operator metrics: %v", err)
		return nil, error
	}
...
}

// InitOperatorService return the static service which expose operator metrics
func InitOperatorService() (*v1.Service, error) {
	operatorName, err := GetOperatorName()
	if err != nil {
		return nil, err
	}
	namespace, err := GetWatchNamespace()
	if err != nil {
		return nil, err
	}
...
}
```

Right now we just log the error and return nil to keep it as best effort.
@LiliC So we either don't return an error in that case to not fail on `operator-sdk up local` :
```Go
	service, err := k8sutil.InitOperatorService()
	if err != nil {
		logrus.Errorf("failed to initialize service object for operator metrics: %v", err)
		return nil, nil
	}
```
Or we pass in the env var `OPERATOR_NAME` in `operator-sdk up local` to avoid that error in the local case.

Reply:
I would say we just keep the `InitOperatorService ` and rename it to `GenerateOperatorMetricService` and return the `error` and the `Service` and let the user handle it in the operator, e.g. the same as with the `GenerateServiceMonitor`.

I would do that in a follow up PR if thats okay with you, along with fixing `operator-sdk up local`.

Reply:
@LiliC Yeah that sounds good. Let's rename and add the fix in a follow up.